jcl/src/java.base/share/classes/java/lang/J9VMInternals.java

```diff
@@ -123,16 +122,18 @@ private static void completeInitialization() {
 					throw new InternalError(e);
 				}
 
-				threads = new Thread[threadGroup.numThreads];
+				threads = new Thread[threadGroup.activeCount()];
 				threadGroup.enumerate(threads, false);
-				/*[ELSE] JAVA_SPEC_VERSION < 19 */
-				threads = Thread.getAllThreads();
-				/*[ENDIF] JAVA_SPEC_VERSION < 19 */
 
 				CleanerShutdown.shutdownCleaner();
 
 				for (Thread t : threads) {
-					if (t.getName().equals("Common-Cleaner")) { //$NON-NLS-1$
+					String threadName = t.getName();
+					if (threadName.equals("Common-Cleaner") //$NON-NLS-1$
+					/*[IF JAVA_SPEC_VERSION >= 26]*/
+							|| threadName.startsWith("InnocuousThread-") //$NON-NLS-1$
+					/*[ENDIF] JAVA_SPEC_VERSION >= 26 */
```

Comment:
Why should this not apply to all versions? `InnocuousThread` exists in all versions.

Reply:
Updated it, along with https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/1137